module/sequencer.cpp

```diff
@@ -162,6 +163,23 @@ void Sequencer::ProcessMultiHomeBatch(Request&& req) {
         lock_only_metadata->insert({key_value.first, metadata.at(key_value.first)});
       }
     }
+
+#ifdef REMASTER_PROTOCOL_COUNTERLESS
+    lock_only_txn->mutable_remaster()->set_new_master((txn.remaster().new_master()));
+    // Add additional lock only at new replica
+    // TODO: refactor to remote metadata from lock-onlys. Requires
+    // changes in the scheduler
+    if (txn.procedure_case() == Transaction::kRemaster) {
+      if (txn.remaster().new_master() == local_rep) {
+        auto key_value = *txn.write_set().begin();
+        lock_only_txn->mutable_write_set()->insert(key_value);
+        auto& new_metadata = (*lock_only_metadata)[key_value.first];
+        new_metadata.set_master(metadata.at(key_value.first).master());
```

Comment:
```C++
lock_only_txn->CopyFrom(txn)
```
Then move the lines for `set_id` and `set_type` above to down below this section